src/form/input/Input.tsx

```diff
@@ -49,32 +51,61 @@ export type InputProps = Omit<
   customClassName?: string;
   inputContainerRef?: React.RefObject<HTMLDivElement>;
   onChange: React.ReactEventHandler<HTMLInputElement>;
-  maxFractionDigits?: number;
+  localizationOptions?: {
+    shouldFormatToLocaleString?: boolean;
+    locale?: string;
+    maximumFractionDigits?: number;
+  };
 };
 
 function Input(props: InputProps) {
   const {
     testid,
+    value,
     type = "text",
     isDisabled,
     hasError,
     customClassName,
     leftIcon,
     rightIcon,
+    localizationOptions = {},
     role,
     autoComplete = "off",
     autoCorrect = "off",
     inputContainerRef,
     onChange,
-    maxFractionDigits = 0,
     ...rest
   } = props;
+  const {
+    shouldFormatToLocaleString = false,
+    locale,
+    maximumFractionDigits = 0
+  } = localizationOptions;
+  const {DECIMAL_NUMBER_SEPARATOR} = getNumberSeparators(locale);
   const inputContainerClassName = classNames("input-container", customClassName);
   const inputClassName = classNames("input", {
     "input--is-disabled": isDisabled,
     "input--has-error": hasError
   });
   const isNumberInput = type === "number";
+  let finalValue = value;
+
+  if (isNumberInput && value && shouldFormatToLocaleString) {
+    finalValue = formatNumber({
+      maximumFractionDigits,
+      locale
+    })(Number(value));
+
+    if (
+      IS_LAST_CHARACTER_DECIMAL_POINT_REGEX.test(String(value)) ||
+      MATCH_ZEROS_AFTER_DECIMAL_REGEX.test(String(value))
+    ) {
+      const decimalPart = String(value).split(DECIMAL_NUMBER_SEPARATOR)[1] || "";
+      const integerPart = finalValue.split(DECIMAL_NUMBER_SEPARATOR)[0];
+
+      finalValue = `${integerPart}${DECIMAL_NUMBER_SEPARATOR}${decimalPart}`;
+    }
```

Comment:
whats this for ?

Reply:
`formatNumber` expects a number for value and we convert it to number. (line: 97)

Meanwhile `decimal separator` and zeros disappear. I think we should hold these. 
```js
const stringNum = "100.0";
const convertedNum = Number(stringNum); // 100 as number
```

Reply:
ha okay makes sense

Reply:
Unresolving this because I have another question:

So let's say the `value` is scientific notation "12345.0" at the moment because user eventually wants to write "12345.005" into the input. To call `formatNumber`, we run `Number("12345.0")` and then `formatNumber` returns us `12.345` (in the TR locale). However, we want the input to read `12.345,0` because the value is supposed to be `"12345.0"`. To achieve this, we run this if block:

```
    if (
      IS_LAST_CHARACTER_DECIMAL_POINT_REGEX.test(String(value)) ||
      MATCH_ZEROS_AFTER_DECIMAL_REGEX.test(String(value))
    ) {
      const decimalPart = String(value).split(DECIMAL_NUMBER_SEPARATOR)[1] || "";
      const integerPart = finalValue.split(DECIMAL_NUMBER_SEPARATOR)[0];

      finalValue = `${integerPart}${DECIMAL_NUMBER_SEPARATOR}${decimalPart}`;
    }
 ```
 
 now the finalValue becomes `12345,0` in the TR locale. However, it should be `12.345,0`, should it not? Did I miss something ?

Reply:
Great, thank you!